plugin/manager.go

```diff
@@ -20,82 +22,73 @@ func init() {
 
 // Plugin is a generic struct for storing info about a plugin.
 type Plugin struct {
-	Path          string
-	Client        *goplugin.Client
-	Reacter       Reacter
-	Subscriptions []Subscription
+	Path    string
+	Reacter Reacter
 }
 
 // Manager handles lifecycle of plugin management.
 type Manager struct {
-	Plugins []*Plugin
-	Log     *zap.Logger
+	Reacters []*Plugin
+	Log      *zap.Logger
 }
 
 // NewManager creates new Manager.
-func NewManager(paths []string, log *zap.Logger) *Manager {
-	plugins := []*Plugin{}
+func NewManager(paths []string, log *zap.Logger) (*Manager, error) {
+	reacters := []*Plugin{}
 	logger := Hclog2ZapLogger{Zap: log}
+
 	for _, path := range paths {
 		client := goplugin.NewClient(&goplugin.ClientConfig{
-			HandshakeConfig: handshakeConfig,
+			HandshakeConfig: shared.Handshake,
 			Plugins:         pluginMap,
 			Cmd:             exec.Command(path),
 			Logger:          logger.Named("PluginManager"),
+			Managed:         true,
 		})
 
-		plugins = append(plugins, &Plugin{
-			Client: client,
-			Path:   path,
-		})
-	}
-
-	return &Manager{
-		Plugins: plugins,
-		Log:     log,
-	}
-}
-
-// Connect connects to plugins.
-func (m *Manager) Connect() error {
-	for _, plugin := range m.Plugins {
-		rpcClient, err := plugin.Client.Client()
+		rpcClient, err := client.Client()
 		if err != nil {
-			return err
+			return nil, err
 		}
 
 		// Request the plugin
-		raw, err := rpcClient.Dispense("subscriber")
+		raw, err := rpcClient.Dispense("reacter")
 		if err != nil {
-			return err
+			return nil, err
 		}
 
-		plugin.Reacter = raw.(*Subscriber)
-		plugin.Subscriptions = plugin.Reacter.Subscriptions()
+		switch raw.(type) {
+		case *ReacterClient:
+			reacters = append(reacters, &Plugin{
+				Path:    path,
+				Reacter: raw.(*ReacterClient),
+			})
```

Comment:
I would still add the default case here, but have it return straight from the switch case instead of panic.

```
switch raw.(type) {
    case *ReacterClient:
        ...
    default:
        return nil, fmt.Errorf("received unexpected type for plugin: %T", raw)
}
```

Reply:
I will change that switch statement in upcoming PRs